python/cog/server/helpers.py

```diff
@@ -361,12 +378,268 @@ def unwrap_pydantic_serialization_iterators(obj: Any) -> Any:
             return [unwrap_pydantic_serialization_iterators(value) for value in obj]
         return obj
 
+else:
+
+    def update_nullable_optional(openapi_schema: Dict[str, Any], app: FastAPI) -> None:
+        def patch_nullable_parameters(openapi_schema: Dict[str, Any]) -> None:
+            for _, methods in openapi_schema.get("paths", {}).items():
+                for _, operation in methods.items():
+                    for param in operation.get("parameters", []):
+                        # If the parameter is optional (required: false), make it nullable
+                        if not param.get("required", True):
+                            schema = param.get("schema", {})
+                            if "nullable" not in schema:
+                                schema["nullable"] = True
+
+        def patch_nullable_union_outputs(openapi_schema: Dict[str, Any]) -> None:
+            for _, schema in (
+                openapi_schema.get("components", {}).get("schemas", {}).items()
+            ):
+                # Look for anyOf with more than one entry
+                if (
+                    "anyOf" in schema
+                    and isinstance(schema["anyOf"], list)
+                    and len(schema["anyOf"]) > 1
+                ):
+                    # If it's missing nullable, and it's meant to represent an Optional/Union output
+                    if "nullable" not in schema:
```

Comment:
Shouldn't this be checking for `null` in the types or something? How do we know that just because it's a union of 2 or more types that it's nullable?

Reply:
I've reworked this so its simpler, now each nullable is the result of a type check: [9e17599](https://github.com/replicate/cog/pull/2316/commits/9e17599d5591ef1932515c8a1b359301b8873dae)